docs/CMakeLists.txt

```diff
@@ -40,12 +60,25 @@ if(NOT QDOC_EXECUTABLE
    OR NOT QDOC_INDEX_DIR
    OR NOT QTATTRIBUTIONSSCANNER_EXECUTABLE
 )
+    # Expand imported target data for status reporting
+    if (TARGET Qt::qdoc)
+        get_target_property(QDOC_EXECUTABLE Qt::qdoc IMPORTED_LOCATION)
+    endif()
+    if (TARGET Qt::qtattributionsscanner)
+        get_target_property(QTATTRIBUTIONSSCANNER_EXECUTABLE Qt::qtattributionsscanner IMPORTED_LOCATION)
+    endif()
+    foreach (_var QDOC_INDEX QDOC_TEMPLATE)
+        if (NOT DEFINED ${_var})
+             set(${_var} "${_var}_DIR=${${_var}_DIR}")
+        endif()
+    endforeach()
+
     message(STATUS "Unable to build user manual in qch format.")
     message(STATUS "qdoc executable: ${QDOC_EXECUTABLE}")
     message(STATUS "qhelpgenerator executable: ${QHELPGEN_EXECUTABLE}")
     message(STATUS "qtattributionsscanner executable: ${QTATTRIBUTIONSSCANNER_EXECUTABLE}")
-    message(STATUS "qdoc template dir: ${QDOC_TEMPLATE_DIR}")
-    message(STATUS "qdoc index dir: ${QDOC_INDEX_DIR}")
+    message(STATUS "qdoc template: ${QDOC_TEMPLATE}")
+    message(STATUS "qdoc index: ${QDOC_INDEX}")
```

Comment:
I slightly went back on my previous change, here, and switched back to displaying the `QDOC_TEMPLATE` and `QDOC_INDEX` (renamed from `QDOC_QTCORE_INDEX`) variables rather than the directory variables. That's so that, if the `find_program()` command is used and fails to find a suitable binary, `QDOC_TEMPLATE-NOTFOUND` or `QDOC_INDEX-NOTFOUND` will be displayed as the value, rather than an empty string.

The overly-complicated bit of code following "# Expand imported target data for status reporting", above, ensures that the variables being reported have sensible values, in the cases where targets are used instead of fallback `find_program()` calls.

Reply:
All of this only ever runs in cases where some part of the help-tool-discovery logic _fails_, so hopefully most of the time it won't even be triggered.